client/galaxy/scripts/components/Upload/config.js

```diff
@@ -0,0 +1,14 @@
+import { getGalaxyInstance } from "app";
+import { getAppRoot } from "onload";
+
+export function initializeUploadDefaults(propsData = {}) {
+    const Galaxy = getGalaxyInstance();
+    const appRoot = getAppRoot();
+    return Object.assign(propsData, {
+        uploadPath: Galaxy.config.nginx_upload_path || `${appRoot}api/tools`,
```

Comment:
I haven't tried it yet, but can this be functional default props in the component, simplifying how this is all spread out?